app/models/person.rb

```diff
@@ -129,6 +129,27 @@ def person
     self
   end
 
+  # method called from the related_items.rb
+  # Gets all the programmes related to this person object.
+  def related_programmes
+    Programme.where(id: related_programme_ids)
+  end
+
+  # Concatenates both lists and filters out the duplicated ids.
+  def related_programme_ids
+    administrator_programme_ids.concat(programme_ids_related_to_projects).uniq
+  end
+
+  # Returns the ids of the programmes related to this person's projects.
+  def programme_ids_related_to_projects
+    Project.includes(:people).where(people: {id: self.id}).map(& :programme_id)
+  end
+
+  # Gets all programme id's who has an programme administrator id equal to this id.
+  def administrator_programme_ids
+    Programme.includes(:programme_administrators).where(programme_administrators: {id: self.id}).map(& :id)
+  end
```

Comment:
you don't actually need all this. you just need `def related_programmes` which returns

``` programmes || programmes_for_role(:programme_administrator) ``` 

instead of `programmes_for_role`,  there is also the cleaner `person.administered_programmes`  but this will also return all programmes if the person is an administrator, so you might want to refactor that slightly

Reply:
https://github.com/seek4science/seek/blob/eb1fc7e2cea43763e951fa831894a419636935ff/lib/seek/roles/accessors.rb#L75

Reply:
I realised I made an error, that was meant to be a single `|` , i.e.
```
programmes | programmes_for_role(:programme_administrator)
```

Reply:
`programmes | programmes_for_role(:programme_administrator)` does the trick and is indeed a lot cleaner.
Thanks!

I've also pushed a unit test and a functional test.